timeboost-sequencer/src/decrypt.rs

```diff
@@ -241,86 +230,133 @@ impl Decrypter {
+            trace!(node = %self.label, committee = %self.current, "dkg bundle already submitted");
             return None;
         }
-        let stores = self.dkg_stores.read();
-        let Some(store) = stores.iter().find(|s| s.committee().id() == committee_id) else {
-            warn!(node = %self.label, committee = %committee_id, "missing dkg store");
+        let guard = self.key_stores.read();
+        let Some(store) = guard.get(self.current) else {
+            warn!(node = %self.label, committee = %self.current, "missing current key store");
+            return None;
+        };
+        let Some(node_idx) = store.committee().get_index(&self.label) else {
+            warn!(node = %self.label, committee = %self.current, "local key not in store for dkg");
             return None;
         };
-        let vess = Vess::new_fast();
 
+        let vess = Vess::new_fast();
         let mut rng = thread_rng();
         let secret = <Vss as VerifiableSecretSharing>::Secret::rand(&mut rng);
         let (ct, cm) = vess
-            .encrypt_shares(committee, store.sorted_keys(), secret, b"dkg")
+            .encrypt_shares(store.committee(), store.sorted_keys(), secret, DKG_AAD)
+            .ok()?;
+        self.submitted.insert(self.current);
+        Some(DkgBundle::new((node_idx, self.label), self.current, ct, cm))
+    }
+
+    /// Generates a resharing bundle for the given key store, if not already submitted.
+    ///
+    /// # Returns
+    /// - `Some(DkgBundle)` if a Resharing dealing was successfully created.
+    /// - `None` if already submitted or if encryption key is missing.
+    pub fn gen_resharing_bundle(&mut self, next_store: KeyStore) -> Option<DkgBundle> {
+        let committee_id = next_store.committee().id();
+        if self.submitted.contains(&committee_id) {
+            trace!(node = %self.label, committee = %committee_id, "resharing bundle already submitted");
+            return None;
+        }
+        let guard = self.key_stores.read();
+        let Some(current_store) = guard.get(self.current) else {
+            warn!(node = %self.label, committee = %self.current, "missing current key store");
+            return None;
+        };
+
+        let Some(node_idx) = current_store.committee().get_index(&self.label) else {
+            warn!(node = %self.label, committee = %self.current, "local key not in store for resharing");
+            return None;
+        };
+
+        let Some(dec_sk) = self.enc_key.get() else {
+            warn!(node = %self.label, committee = %committee_id, "no existing key to reshare");
+            return None;
+        };
+        let vess = Vess::new_fast();
+        let share = dec_sk.privkey().share();
+        let secret = <Vss as VerifiableSecretSharing>::Secret::from(*share);
+        let (ct, cm) = vess
+            .encrypt_reshares(
+                next_store.committee(),
+                next_store.sorted_keys(),
+                secret,
```

Comment:
```suggestion
        let (ct, cm) = vess
            .encrypt_reshares(
                next_store.committee(),
                next_store.sorted_keys(),
                *dec_sk.privkey().share(),
```
this should work? (no need to convert)

Reply:
Thanks.
Fixed in: dc451dc7